lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -189,25 +189,52 @@ static Value buildUnitNormalCdf(OpBuilder &b, Location &loc, Value x) {
   return buildNormalCdf(b, loc, x, zero, one);
 }
 
+template<typename TorchOpTy, typename MathOpTy>
+static Value createCalculationForMathOpWithDtypeConversion(
+              OpBuilder &b, Location loc, TypeConverter *converter,
+              ValueRange payloadArgs, Operation *op) {
+  auto opCasted = dyn_cast<TorchOpTy>(op);
+  Type dtype = converter->convertType(opCasted.getType())
+                    .template cast<RankedTensorType>()
+                    .getElementType();
+  Value arg = convertScalarToDtype(b, loc, payloadArgs[0], dtype);
+  return b.create<MathOpTy>(loc, arg);
+}
+
 static Value createLinalgPayloadCalculationForElementwiseOp(
     OpBuilder &b, Location loc, TypeConverter *converter,
     ValueRange payloadArgs, Operation *op, ArrayRef<Value> operands) {
-  if (isa<AtenTanhOp>(op))
-    return b.create<math::TanhOp>(loc, payloadArgs[0]);
-  if (isa<AtenExpOp>(op))
-    return b.create<math::ExpOp>(loc, payloadArgs[0]);
   if (isa<AtenFloorOp>(op))
     return b.create<math::FloorOp>(loc, payloadArgs[0]);
   if (isa<AtenCeilOp>(op))
     return b.create<math::CeilOp>(loc, payloadArgs[0]);
-  if (isa<AtenLogOp>(op))
-    return b.create<math::LogOp>(loc, payloadArgs[0]);
-  if (isa<AtenErfOp>(op))
-    return b.create<math::ErfOp>(loc, payloadArgs[0]);
-  if (isa<AtenSqrtOp>(op))
-    return b.create<math::SqrtOp>(loc, payloadArgs[0]);
+  if (isa<AtenTanhOp>(op)) 
```

Comment:
nit: Here we follow the style convention of adding curly braces to any `if` with a body more than one line long